### PR TITLE
Add Sprite changes to Logic Gates to show the input/output state

### PR DESCRIPTION
**Pull request by 0tito**
## About the PR
Gates now show the state of their inputs and outputs on the sprite

## Why / Balance
Its easier to use them if you can see the states

## Media
https://github.com/user-attachments/assets/c305a2b9-adcb-420f-8a2e-f5ad60a11166

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

**Changelog**

### CODE REVIEW

**Comment by deltanedas** [comment]
use separate layers for a/b/output and none of the evil stuff is needed

**Comment by 0tito** [comment]
Is this better? im not exactly the best coder :godo:

**Review by Aeshus — CHANGES_REQUESTED** [review]
Thanks for the PR!

There's some small code style issues, and a bug where the code accidentally sets InputA instead of InputB.

Other than that, I like the change and the rest of the visualizer/layering stuff looks good (I don't have personal experience with any of it).

Also, I'm not a maintainer, so this is just a peer review. I may have made some mistakes in my review, or have missed things, so take this review with a grain of salt.

File: Content.Shared/DeviceLinking/SharedLogicGate.cs
Comment (deltanedas) [inline Content.Shared/DeviceLinking/SharedLogicGate.cs:45]: copy paste of SignalState

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Review by Aeshus — APPROVED** [review]
LGTM! Thanks for applying the changes/feedback.

I think the only thing now is that you could add yourself to the `meta.json`'s copyright to say that you created the sprites for the light-up states.

Also, make sure to click "resolve" for comments/requested changes once you address it. Otherwise, maintainers may have the false impression that it is still a work-in-progress or ongoing issue.

File: Content.Server/DeviceLinking/Systems/LogicGateSystem.cs
Comment (slarticodefast) [inline Content.Server/DeviceLinking/Systems/LogicGateSystem.cs:105]: Instead of using the strings "High" and "Low" you can simply make it a boolean

File: Resources/Prototypes/Entities/Structures/gates.yml
Comment (slarticodefast) [inline Resources/Prototypes/Entities/Structures/gates.yml:39]: instead of using an empty sprite just toggle the layers on or off

**Review by slarticodefast — CHANGES_REQUESTED** [review]
Cool idea. Needs a little more cleanup before it can be merged.
Thanks to Aeshus and delta for doing the previous reviews.

**Comment by slarticodefast** [comment]
Quick question: What changes did you make to logic.png? I don't see any difference yet somehow it still doubled in size.

**Comment by 0tito** [comment]
I used it as base when i was making the other sprites, and i think i changed its sprite when  i was learning how to change sprites but then instead of reverting it i painted it back to a similar color :godo: i used krita, maybe when i saved i changed something on it?

**Comment by slarticodefast** [comment]
If it is not changed leave it as it was to preserve the git history and attributions.
And I gave the PR a quick in-game test and the sprites for gates other than the or gate seem to be wrong now when spawning them.
grafik

**Comment by 0tito** [comment]
This happens in normal wizden too, iirc oasis has some gates like that, but if you craft them and  switch through them normally they work. I can try to change it tho, also how do i "revert" the change to logic.png, would just replacing the sprite with upstream logic.png fix it? or do i have to do something else

**Comment by slarticodefast** [comment]
Just copying the original sprite back is probably the easiest way. There should be a git command, too, but no idea which one.
Thanks for fixing the other gates! I'll give it another test later.

**Review by slarticodefast — APPROVED** [review]
I tested this and it works as intended for non-pulsed signals. With signal buttons there seems to be some weird behaviour for gates, but that is not the fault of this PR and happening on master as well. I opened an issue for that #33716.
Thank you for your first contribution!

**Pull request merged**
